Battle/Field/Friendlies.gd

```diff
@@ -2,26 +2,35 @@ extends Control
 # This file is very similar to enemies.gd.
 # They should probably be merged at some point!
 
+#@export var current_target: int = 0
+
 # Called when the node enters the scene tree for the first time.
 func _ready():
 	pass # Replace with function body.
 
+func set_speed(new_speed: float):
+	# loop through each unit and set its speed 
+	for child in get_children():
+		child.speed = new_speed
+		
 func get_random_target():
 	var children = get_children()
+	var battleNode = get_parent()
+	var unitArray = battleNode.units
```

Comment:
Similar to the other comment, it would be better if we didn't have to check a parent's node properties. Since we only use it to check if the specified unit is dead, it might be helpful to just set an `is_dead` property on the `field_unit` itself. It would also be helpful for knowing when to start a death animation. `friendlies` would check that property in this loop instead of checking a parent node.

However, since we will most likely always have the friendlies node within the battle scene, it is probably fine.

Reply:
Taking care in the next Commit.

Commenting the code ⏳